conduit/pipeline/pipeline_test.go

```diff
@@ -249,59 +267,84 @@ func mockPipeline(t *testing.T, dataDir string) (*pipelineImpl, *test.Hook, *moc
 
 // TestPipelineRun tests that running the pipeline calls the correct functions with mocking
 func TestPipelineRun(t *testing.T) {
-	mImporter := mockImporter{}
-	mImporter.On("GetBlock", mock.Anything).Return(uniqueBlockData, nil)
+	mImporter := mockImporter{rndOverride: initialRound + 1}
+	mImporter.On("GetBlock", mock.Anything).Return(mock.Anything, nil)
+	mImporter.On("OnComplete", mock.Anything).Return(nil)
 	mProcessor := mockProcessor{}
 	processorData := uniqueBlockData
 	processorData.BlockHeader.Round++
-	mProcessor.On("Process", mock.Anything).Return(processorData)
+	mProcessor.On("Process", mock.Anything).Return(mock.Anything)
 	mProcessor.On("OnComplete", mock.Anything).Return(nil)
 	mExporter := mockExporter{}
 	mExporter.On("Receive", mock.Anything).Return(nil)
+	mExporter.On("OnComplete", mock.Anything).Return(nil)
 
 	var pImporter importers.Importer = &mImporter
 	var pProcessor processors.Processor = &mProcessor
 	var pExporter exporters.Exporter = &mExporter
-	var cbComplete conduit.Completed = &mProcessor
 
-	ctx, cf := context.WithCancel(context.Background())
+	ctx, ccf := context.WithCancelCause(context.Background())
 
 	l, _ := test.NewNullLogger()
 	pImpl := pipelineImpl{
-		ctx:              ctx,
-		cf:               cf,
-		logger:           l,
-		initProvider:     nil,
-		importer:         pImporter,
-		processors:       []processors.Processor{pProcessor},
-		exporter:         pExporter,
-		completeCallback: []conduit.OnCompleteFunc{cbComplete.OnComplete},
+		ctx:          ctx,
+		ccf:          ccf,
+		logger:       l,
+		initProvider: nil,
+		importer:     pImporter,
+		processors:   []processors.Processor{pProcessor},
+		exporter:     pExporter,
 		pipelineMetadata: state{
 			NextRound:   0,
 			GenesisHash: "",
 		},
 		cfg: &data.Config{
 			RetryDelay: 0 * time.Second,
 			RetryCount: math.MaxUint64,
+			Processors: []data.NameConfigPair{{
+				Name:   "mockProcessor",
+				Config: map[string]interface{}{},
+			}},
```

Comment:
This was getting an index-out-of-bounds panic because I introduced `Init()` on line 320 below. Then inside of `Init()` there's [an expectation](https://github.com/tzaffi/conduit/blob/4f5576352b6cad9eee190883f5af0b4dbe1e2cc4/conduit/pipeline/pipeline.go#L351) that the pipeline's  config has a slice storing the processor configs.